src/containers/home/Events.jsx

```diff
@@ -76,33 +77,43 @@ const Events = () => {
                 </div>
               ) : (
                 listToShow.map(
-                  ({ location, date, guests, name, link }, index) => {
+                  ({ location, date, guests, name, link, isPast }, index) => {
                     const info = [
                       `${dayjs(date).format('MMM DD, YYYY')} at ${dayjs(
                         date
-                      ).format('hh:mm a')}`,
-                      name
+                      ).format('hh:mm a')}`
                     ];
                     if (guests > 0) {
                       info.push(`${guests} guests`);
                     }
                     return (
                       <div
                         key={link}
-                        className={classnames('py-3', {
+                        className={classnames('py-3 relative', {
                           'border-b border-gray-200':
-                            index < fortcoming.length - 1
+                            index < listToShow.length - 1
                         })}
                       >
-                        <a
-                          href={link}
-                          target="_blank"
-                          rel="noopener noreferrer"
-                        >
-                          <H6 className="hover:text-primary transition-colors duration-200">
-                            {name}
-                          </H6>
-                        </a>
+                        {isPast && (
+                          <div className="absolute h-full w-full bg-foreground opacity-50 z-10 pointer-events-none" />
+                        )}
+                        <div className="flex pt-2">
+                          <a
+                            style={{ maxWidth: 'calc(100% - 5rem)' }}
+                            href={link}
+                            target="_blank"
+                            rel="noopener noreferrer"
+                          >
+                            <H6 className="hover:text-primary transition-colors duration-200">
+                              {name}
+                            </H6>
+                          </a>
+                          {isPast && (
+                            <span className="w-20 relative opacity-90 z-50 pointer-events-none">
```

Comment:
I think it would be fine without lowering the opacity
```suggestion
                            <span className="w-20 relative z-50 pointer-events-none">
```